pmlc/tools/pmlc-opt/BUILD

```diff
@@ -0,0 +1,28 @@
+# Copyright 2019 Intel Corporation.
+
+package(default_visibility = ["//visibility:public"])
+
+load("//bzl:plaidml.bzl", "plaidml_cc_binary")
+
+plaidml_cc_binary(
```

Comment:
I think it would be useful to enable other compilers built on this technology.  For example, I think it would be useful to split `pmlc-opt` into two targets: a library containing all of these dependencies and the current `main()` entrypoint, and a trivial cc_binary that links them all together.  If `lit.bzl` and friends were also changed to take their `pmlc-opt` target as an input, that would make it easy for other compilers using these components to leverage the same lit test infrastructure, instead of having to roll their own.

(We could also do that in a separate PR -- if the basic idea sounds okay, LMK, and I'll go ahead and put it together after the current PR lands.)

Reply:
For lit testing, you're going to need to make your own `lit.site.py` and whatnot to use your own compiler. I'm not sure what value we'd get in having a `pmlc-opt-lib`, seems like we can just have any other custom compilers bring in the pieces they need and write their own custom main as needed.

Reply:
Yeah, that's basically what I did in another project -- I copied the PlaidML files and tweaked them to fit the other project's situation, but it seemed like suboptimal technical debt to be cleaned up at some point, not the right way forward.

Reply:
Happy to see any proposals to improve the situation, I don't have a concrete feel for what change we need to make at this time.